code/game/machinery/mindswapper.dm

```diff
@@ -0,0 +1,98 @@
+
+/obj/machinery/mindswapper
+	name = "experimental mind swapper"
+	desc = "The name isn't descriptive enough?"
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grinder"
+	density = TRUE
+	anchored = TRUE
+	layer = BELOW_OBJ_LAYER
+	// req_access = list(access_kitchen,access_morgue)
+	circuit = /obj/item/weapon/circuitboard/mindswapper
+
+	var/operating = FALSE  // Is it on?
+	var/swap_time = 200  // Time from starting until minds are swapped
+	var/swap_range = 1
+
+	use_power = 1
+	idle_power_usage = 2
+	active_power_usage = 500
+
+/obj/machinery/mindswapper/update_icon()
+	overlays.Cut()
+	if(stat & (NOPOWER|BROKEN))
+		return
+	/*if (operating) // Overlay when the mindswapper is active
+		overlays += image('icons/obj/kitchen.dmi', "gruse")*/
+
+/obj/machinery/mindswapper/attack_hand(mob/user as mob)
+	if(stat & (NOPOWER|BROKEN))
+		return
+	if(operating)
+		to_chat(user, SPAN_DANGER("The mind swapping process has been launched, there is no going back now."))
+		return
+	else
+		startswapping(user)
+
+/obj/machinery/mindswapper/attackby(obj/item/I, mob/user)
+	..()
+
+/obj/machinery/mindswapper/examine()
+	..()
+	to_chat(usr, "The safety is [emagged ? SPAN_DANGER("disabled") : "enabled"].")
+
+/obj/machinery/mindswapper/emag_act(var/remaining_charges, var/mob/user)
+	emagged = !emagged
+	to_chat(user, SPAN_DANGER("You [emagged ? "disable" : "enable"] the mind swapper safety."))
+	if(emagged)
+		swap_time = 50
+	else
+		swap_time = 200
+	return 1
+
+/obj/machinery/mindswapper/proc/startswapping(mob/user as mob)
+	if(operating)
+		return
+
+	use_power(1000)
+	visible_message(SPAN_DANGER("You hear an increasingly loud humming coming from the mind swapper."))
+	operating = TRUE
+	update_icon()
+
+	user.attack_log += "\[[time_stamp()]\] Triggered the mind swapper</b>"
+	msg_admin_attack("[user.name] ([user.ckey]) triggered the mind swapper (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)")
+
+	spawn(swap_time)
```

Comment:
Move spawn to timer
`addtimer(CALLBACK(src, .proc/foo), swap_time, TIMER_STOPPABLE))`